### DISCUSSION TITLE
run on startup? Raspberry Pi

### DISCUSSION THREAD

**Discussion by sacoward** [tgorgdotcom/locast2plex#160] [Q&A]
Is there any way to automatically run this on startup on a Raspberry Pi?  I am very new to this.  I was able to get it working without a problem, by just downloading the files, editing the config.ini, & running pythong3 main.py on the folder.  Do I have to do that (the last part) every time I restart by Raspberry Pi?

Thanks!

**Comment by CTJohnK** [comment]
Edit rc.local in /etc/rc.local and add your command: 
- make sure to add a  '&' at the end of the line!
- make sure to make locast folder .py files chmod 777!
- make sure rc.local is executable!

**Reply by ajderder** [reply]
Hi, I have tried this and am not able to get it running at bootup. This is the rc.local file I am using:

#!/bin/sh -e

_IP=$(hostname -I) || true
if [ "$_IP" ]; then
  printf "My IP address is %s\n" "$_IP"
fi

su - pi -c '/usr/bin/tightvncserver :1 -geometry 1920x1080 ' &

python3  /home/pi/locast2plex/main.py &
exit 0

Any advice would greatly be appreciated!
Thanks!

**Reply by ajderder** [reply]
I have assured that the folder and .py file are executable as well

**Comment by CTJohnK** [comment]
I'm still using an older version of L2P which is why it includes user and password (blanked), but I've attached a screenshot of my rc.local.  I presume you are running this out of /etc, correct?

For clarity, when you say "can't see it running on startup" do you mean Plex does not see L2P and you cannot connect to it?  Also I did include "sudo" as a prefix to my command.

![Capture](https://user-images.githubusercontent.com/69587153/110523382-0895d980-80e0-11eb-8b59-20f1321a1cf9.JPG)

**Reply by ajderder** [reply]
Thanks for your reply!! Please see my comments below

I presume you are running this out of /etc, correct? Yes that is correct
'
For clarity, when you say "can't see it running on startup" do you mean Plex does not see L2P and you cannot connect to it?  Yes this is correct

I also added sudo to the command with no luck

**Comment by CTJohnK** [comment]
To confirm, if you run the command from the command prompt, Plex can connect, correct?On Mar 9, 2021 7:12 PM, ajderder ***@***.***> wrote:
Thanks for your reply!! Please see my comments below
I presume you are running this out of /etc, correct? Yes that is correct
'
For clarity, when you say "can't see it running on startup" do you mean Plex does not see L2P and you cannot connect to it?  Yes this is correct
I also added sudo to the command with no luck
